### PR TITLE
Remove redundant use of `await` on a return value

### PR DESCRIPTION
**Pull request by apupier**
as reported as code smell by Sonar
https://sonarcloud.io/project/issues?id=vscode-camelk&resolved=false&rules=typescript%3AS4326&types=CODE_SMELL

Signed-off-by: Aurélien Pupier <apupier@redhat.com>

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -160,7 +160,7 @@ export function getKamelRuntimeVersionConfig(): string | undefined {
 
 export async function setKamelRuntimeVersionConfig(value : string) : Promise<void> {
 	const configuration = vscode.workspace.getConfiguration();
-	return await configuration.update(RUNTIME_VERSION_KEY, stripLeadingV(value), vscode.ConfigurationTarget.Global);
+	return configuration.update(RUNTIME_VERSION_KEY, stripLeadingV(value), vscode.ConfigurationTarget.Global);
```
Comment (bfitzpat) [inline src/config.ts:163]: If await is no longer needed, do we need to return a Promise?

File: src/config.ts [hunk 2]
```diff
@@ -171,5 +171,5 @@ export function getKamelAutoupgradeConfig() : boolean {
 // for testing purposes 
 export async function setKamelAutoupgradeConfig(value : boolean) : Promise<void> {
 	const configuration = vscode.workspace.getConfiguration();
-	return await configuration.update(AUTOUPGRADE_KEY, value, vscode.ConfigurationTarget.Global);
+	return configuration.update(AUTOUPGRADE_KEY, value, vscode.ConfigurationTarget.Global);
```
Comment (bfitzpat) [inline src/config.ts:174]: If await is no longer needed, do we need to return a Promise?

File: src/kamel.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ class KamelImpl implements Kamel {
 	}
 
 	async getPath(): Promise<string> {
-		return await baseKamelPath();
+		return baseKamelPath();
```
Comment (bfitzpat) [inline src/kamel.ts:44]: If await is no longer needed, do we need to return a Promise?

File: src/kubectl.ts [hunk 4]
```diff
@@ -39,7 +39,7 @@ class KubectlImpl implements Kubectl {
 	}
 
 	async getPath(): Promise<string> {
-		return await baseKubectlPath();
+		return baseKubectlPath();
```
Comment (bfitzpat) [inline src/kubectl.ts:42]: If await is no longer needed, do we need to return a Promise?

**Review by bfitzpat — COMMENTED** [review]
So for all of these methods where we're getting rid of await, do we need to return a Promise at all?

File: src/config.ts (hunk 1 above)
Comment (apupier) [reply src/config.ts:163]: the `await` is not needed for returned values inside the `async` method which is returning a `Promise`.

more information in the description of the issue on Sonar: https://sonarcloud.io/organizations/camel-tooling/rules?open=typescript%3AS4326&rule_key=typescript%3AS4326

**Review by bfitzpat — APPROVED** [review]

**Pull request merged**